ui/assets/js/signal.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Signal represents a collection of observers that can be notified with
+ * parameterized event dispatch. Listeners can tap into a signal to
+ * receive callbacks when a signal is raised.
+ */
+export class Signal {
+	constructor() {
```

Comment:
We technically use `prettier` configured to a tab space of 4, although it doesn't seem like we enforce that in CI. 
https://github.com/hound-search/hound/blob/main/.prettierrc

Reply:
Ah. I ran `npm run format` but I think that only works on files that are staged in git?